demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -92,6 +90,143 @@ def _is_incident_type_default_playbook_found(self, incident_type_data):
+        scripts_set = set(get_layouts_scripts_ids(layouts_container_tabs))
+
+        # Check if the layouts container's scripts are in the id_set:
+        if scripts_set:
+            scripts_not_in_id_set = self._get_scripts_that_are_not_in_id_set(scripts_set)
+
+        # Add error message if there are scripts that aren't in the id_set:
+        if scripts_not_in_id_set:
+            is_valid = False
+            scripts_not_in_id_set_str = ', '.join(scripts_set)
+            error_message, error_code = Errors.layouts_container_non_existent_script_id(layouts_container_name,
+                                                                                        scripts_not_in_id_set_str)
+            if not self.handle_error(error_message, error_code, file_path="id_set.json"):
+                is_valid = True
+
+        return is_valid
+
+    def _is_layout_scripts_found(self, layout_data):
+        """Check if scripts of a layout  is in the id_set
+
+        Args:
+            layout_data (dict): Dictionary that holds the extracted details from the given layout.
+
+        Returns:
+            bool. Whether the scripts are in the id_set or not.
+        """
+        is_valid = True
+        scripts_not_in_id_set = set()
+
+        layout_id = list(layout_data.keys())[0]
+        layout = layout_data.get(layout_id, {})
+        layout_name = layout.get('typename', layout_id)
+        scripts = layout.get('scripts', [])
+        scripts_set = set(scripts)
+
+        # Check if the layouts container's scripts are in the id_set:
+        if scripts_set:
+            scripts_not_in_id_set = self._get_scripts_that_are_not_in_id_set(scripts_set)
+
+        # Add error message if there are scripts that aren't in the id_set:
+        if scripts_not_in_id_set:
+            is_valid = False
+            scripts_not_in_id_set_str = ', '.join(scripts_set)
+            error_message, error_code = Errors.layout_non_existent_script_id(layout_name,
+                                                                             scripts_not_in_id_set_str)
+            if not self.handle_error(error_message, error_code, file_path="id_set.json"):
+                is_valid = True
+
+        return is_valid
+
+    def _get_scripts_that_are_not_in_id_set(self, scripts_set):
+        """
+        For each script ID in the given scripts set checks if it is exist in the id set.
+        If a script is in the id set removes it from the input scripts set.
+
+        Args:
+            scripts_set: A set of scripts IDs
+
+        Returns:
+            A sub set  of the input scripts set which contains only scripts that re not in the id set.
+        """
+        for checked_script in self.script_set:
+            checked_script_id = list(checked_script.keys())[0]
+            if checked_script_id in scripts_set:
+                scripts_set.remove(checked_script_id)
+
+        return scripts_set
```

Comment:
self.script_set is not a dict, it's a list, therefore in order to create the set 'all_script_ids' that includes the id's of all scripts in self.script_set (content repo) we need to perform the following loop: 
```
 for checked_script in self.script_set:
            checked_script_id = list(checked_script.keys())[0]
            all_script_ids.add( checked_script_id)
```

After creating the set we will need to go over the smaller set - scripts_set - and check if it's scripts exist in all_script_ids set. 

Because in creating the set we go through a loop on all the scripts in content repo I think it is unnecessary to perform another search on the set and it is possible to check the existence of the scripts already in the first loop as I implemented.